lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -63,6 +67,18 @@ export class PublicClientApplication extends ClientApplication implements IPubli
      * @returns {Promise.<AuthenticationResult>} - a promise that is fulfilled when this function has completed, or rejected if an error was raised. Returns the {@link AuthResponse} object
      */
     loginPopup(request?: PopupRequest): Promise<AuthenticationResult> {
+        if (this.isBrowserEnvironment) {
+            this.broadcastEvent(BroadcastEvent.LOGIN_START, InteractionType.POPUP, request);
+            return this.acquireTokenPopup(request || DEFAULT_REQUEST)
+                .then((result) => {
+                    this.broadcastEvent(BroadcastEvent.LOGIN_SUCCESS, InteractionType.POPUP, result);
+                    return result;
+                })
+                .catch((e) => {
+                    this.broadcastEvent(BroadcastEvent.LOGIN_FAILURE, InteractionType.POPUP, null, e);
+                    throw e;
```

Comment:
I don't think there needs to be two different blocks, these should be able to hang off of the existing `acquireTokenPopup` invocation below.

Reply:
Cleaned up!